frontend/src/components/pages/CampOverview/CampRegistrationLink.tsx

```diff
@@ -0,0 +1,50 @@
+import {
+  HStack,
+  Input,
+  Button,
+  InputGroup,
+  InputLeftElement,
+  useToast,
+} from "@chakra-ui/react";
+import { LinkIcon } from "@chakra-ui/icons";
+import React from "react";
+
+const CampRegistrationLink = ({
+  p,
+  isDisabled,
+}: {
+  p: string;
+  isDisabled: boolean;
+}): JSX.Element => {
+  const toast = useToast();
+
+  return (
+    <HStack justifyContent="left" marginBottom="8px">
+      <InputGroup backgroundColor="white" pointerEvents="none" size="sm">
+        <InputLeftElement>
+          <LinkIcon color="gray.300" />
+        </InputLeftElement>
+        <Input disabled placeholder={p} borderRadius="0" />
+      </InputGroup>
+      <Button
+        colorScheme="green"
+        borderRadius="0"
+        size="sm"
+        disabled={isDisabled}
```

Comment:
I think the designers wanted the button to be in a "copied" and disabled state temporarily for 3 seconds and then change back to the copy CTA. 

We can handle this by passing in another flag saying isCopied which is first set to false but onclick gets set to true. then we can trigger a 3 second timeout after which the state for that flag is set to copied again. Here are some discussions which might be helpful:
https://upmostly.com/tutorials/settimeout-in-react-components-using-hooks